python-test-samples/integrated-application-test-kit/README.md

```diff
@@ -0,0 +1,112 @@
+* Postvalidate Plugin Sample with the Integrated Application Test Kit
+  * [app.py](./plugins/2-postvalidate-plugins/python-minimal-plugin/functions/app.py)
+  * [test_by_polling.py](./plugins/2-postvalidate-plugins/python-minimal-plugin/tests/integration/test_by_polling.py)
+  * [test_by_waiting.py](./plugins/2-postvalidate-plugins/python-minimal-plugin/tests/integration/test_by_waiting.py)
+  * [test_by_polling_with_registry.py](./plugins/2-postvalidate-plugins/python-minimal-plugin/tests/integration/app.py)
+
+## Sample project description
+
+This sample is built around the idea of a Video Plugin application where users might upload video streams to the platform. Lifecycle events are triggered as the videos are processed. Plugins can react to the events to perform work on the video or video metadata.
+
+![Architecture](./img/architecture.png)
+
+The system includes a [plugin tester](./plugins/plugin_tester/README.md), which is an AWS Step Functions workflow that acts as the video platform - emitting lifecycle events with the expected structure and sequence. [The sample tests](./plugins/2-postvalidate-plugins/python-minimal-plugin/tests/) integrate with the plugin tester to exercise plugin functionality.
+
+![Plugin Tester](./img/plugin_tester_1.png)
+
+## Run the integration tests
+
+### Build and Deploy the plugin tester
+
+Before you run the integration tests, deploy the [plugin tester](./plugins/plugin_tester/).
+
+```bash
+cd ./plugins/plugin_tester && sam build --use-container
+cd ./plugins/plugin_tester && sam deploy --no-confirm-changeset --no-fail-on-empty-changeset
+```
+
+### Run the prevalidate integration tests
+
+The prevalidate plugin is a sample plugin that acts on the [prevalidate event](./events/1-prevalidate-event.md) emitted by the platform. These integration tests use the AWS SDK to query the Amazon CloudFormation API directly and poll the plugin tester workflow.
+
+Deploy the [test-prevalidate-plugin](./plugins/1-prevalidate-plugins/test-prevalidate-plugin/)
+
+```bash
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && sam build --use-container
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && sam deploy --no-confirm-changeset --no-fail-on-empty-changeset
+```
+
+Run the [test-prevalidate-plugin](./plugins/1-prevalidate-plugins/test-prevalidate-plugin/) integration tests
+
+Replace the `AWS_REGION` parameter with the region you deployed to.
+
+```bash
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && npm install
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 npm test
+```
+
+### Run the postvalidate integration tests
+
+The postvalidate plugin is a sample plugin that acts on the [postvalidate event](./events/2-postvalidate-event.md) emitted by the platform. These integration tests use the Integrated Application Test Kit to find cloud-hosted resources and listen for activity.
```

Comment:
In contrast to the previous test example, these tests show how the IATK simplifies some common testing tasks.